crates/audio/src/spatial.rs

```diff
@@ -0,0 +1,182 @@
+    focus: &CameraFocus,
+    sound_position: Vec3,
+) -> (f64, f64) {
+    let cam_right = camera.right();
+    let sound_dir = (sound_position - camera.translation()).normalize();
+    let pan = cam_right.dot(sound_dir) * 0.5 + 0.5;
+
+    // Simulates sounds becoming quieter when further away
+    let attenuation_factor = {
+        let distance_from_camera_squared = camera.translation().distance_squared(sound_position);
+        // Anything closer than 70% of zoom distance is at full volume.
+        let min_distance_squared = (0.7 * focus.distance().inner()).powi(2);
+
+        min_distance_squared / distance_from_camera_squared
+    };
+
+    // Silences sounds whose sources are not in view
+    let occlusion_factor = {
+        // Volume is 1 from 0-25.5 degrees from the view direction, then linearly goes to 0 from 25.5-45 degrees.
+        // Angle is calculated from a point slightly behind the camera.
+        let camera_offset = camera.translation() + camera.back() * SOUND_VIEW_ANGLE_OFFSET;
+        let angle = (sound_position - camera_offset).angle_between(camera.forward());
+
+        // Slope parameter of linear function (-1 unit in 25.5 degrees or pi/8)
+        const SLOPE: f32 = -1.0 / (PI / 8.);
+        // Constant parameter of linear function
+        const CONSTANT: f32 = 2.0;
+        // Let's limit this to non-negative to avoid the weirdness that
+        // occurs if both factors end up being negative at some point
+        (SLOPE * angle + CONSTANT).max(0.0)
+    };
+
+    let volume = (attenuation_factor * occlusion_factor).clamp(0., 1.);
+
+    (volume as f64, pan as f64)
+}
+
+fn play(
+    mut commands: Commands,
+    camera: Query<&GlobalTransform, With<Camera>>,
+    focus: Res<CameraFocus>,
+    audio: Res<KAudio>,
+    sounds: Res<Sounds>,
+    mut play_events: EventReader<PlaySpatialAudioEvent>,
+) {
+    let camera = camera.single();
+
+    for PlaySpatialAudioEvent { sound, position } in &mut play_events {
+        let (volume, pan) = calculate_volume_and_pan(camera, &focus, *position);
+        let handle = audio
+            .play(sounds.0[*sound].clone())
+            .with_volume(volume)
+            .with_panning(pan)
+            .handle();
+
+        commands.spawn((
+            TransformBundle::from_transform(Transform::from_translation(*position)),
+            handle,
+            SpatialSound,
+        ));
+    }
+}
+
+type InitializedSound<'s> = (Entity, &'s Handle<AudioInstance>, &'s GlobalTransform);
+
+fn update_spatial(
+    mut commands: Commands,
+    spatial_audios: Query<InitializedSound, With<SpatialSound>>,
+    camera: Query<&GlobalTransform, With<Camera>>,
+    focus: Res<CameraFocus>,
+    mut audio_instances: ResMut<Assets<AudioInstance>>,
+) {
+    let camera = camera.single();
+
+    for (entity, audio, transform) in &spatial_audios {
+        let Some(audio_instance) = audio_instances.get_mut(audio) else {
+            commands.entity(entity).despawn();
```

Comment:
I would use `despanw_recursive()` so that there are no orphans in case we add any children in the future.